### PR TITLE
Warn about usage of non-breaking space

### PR DESCRIPTION
**Pull request by byara**
#### What changed in this PR:

This PR adds a warning about usage of non-breaking space in a template. Regarding this issue, a unit test is added.

### CODE REVIEW

**Comment by byara** [comment]
This issue could be addressed also by creating a new Token for `non-breaking space` and Melody would automatically throw a warning like, `Invalid Token Expected symbol but found NonBreakingSpace instead.`

**Comment by pago** [comment]
Question: Do we also want to cover other unicode sequences which represent whitespace?

Resources:
https://codepen.io/rubikzube/pen/wolgB
http://xahlee.info/js/js_whitespace.html
https://mathiasbynens.be/notes/javascript-escapes
https://mathiasbynens.be/notes/javascript-unicode

**Comment by byara** [comment]
> Question: Do we also want to cover other unicode sequences which represent whitespace?
> 
> Resources:
> https://codepen.io/rubikzube/pen/wolgB
> http://xahlee.info/js/js_whitespace.html
> https://mathiasbynens.be/notes/javascript-escapes
> https://mathiasbynens.be/notes/javascript-unicode

I believe it is a good idea to address all unicode sequences which represent `whitespace`. It prevent the future bugs/bug reports regarding these type of issues.

We can also follow Babel example and use something similar to `https://github.com/xtuc/charcodes/tree/master/packages/charcodes` and use `charCodeAt`. Or just add the characters and address them in the appropriate function. One benefit of using `charCodeAt` + the separate package would be to target characters that cannot be easily read or shown.

**Comment by pago** [comment]
Do what you think is best. :)

File: packages/melody-parser/src/Lexer.js [hunk 1]
```diff
@@ -355,6 +355,11 @@ export default class Lexer {
                 } else if (CHAR_TO_TOKEN.hasOwnProperty(c)) {
                     input.next();
                     return this.createToken(CHAR_TO_TOKEN[c], pos);
+                } else if (c === ' ') {
```
[the remark this answers is not in the document]
Comment (byara) [reply packages/melody-parser/src/Lexer.js]: That is true, it makes the code more readable. On the other hand, I had something like this in mind, that we could add in melody:

```javascript
export const backSpace = 8;
export const tab = 9; //  '\t'
export const lineFeed = 10; //  '\n'
export const carriageReturn = 13; //  '\r'
export const shiftOut = 14;
export const space = 32;
export const exclamationMark = 33; //  '!'
export const quotationMark = 34; //  '"'
export const numberSign = 35; //  '#'
export const dollarSign = 36; //  '$'
export const percentSign = 37; //  '%'
export const ampersand = 38; //  '&'
export const apostrophe = 39; //  '''
export const leftParenthesis = 40; //  '('
export const rightParenthesis = 41; //  ')'
export const asterisk = 42; //  '*'
export const plusSign = 43; //  '+'
export const comma = 44; //  ','
export const dash = 45; //  '-'
export const dot = 46; //  '.'
export const slash = 47; //  '/'
export const digit0 = 48; //  '0'
export const digit1 = 49; //  '1'
export const digit2 = 50; //  '2'
export const digit3 = 51; //  '3'
export const digit4 = 52; //  '4'
export const digit5 = 53; //  '5'
export const digit6 = 54; //  '6'
export const digit7 = 55; //  '7'
export const digit8 = 56; //  '8'
export const digit9 = 57; //  '9'
export const colon = 58; //  '
export const semicolon = 59; //  ';'
export const lessThan = 60; //  '<'
export const equalsTo = 61; //  '='
export const greaterThan = 62; //  '>'
export const questionMark = 63; //  '?'
export const atSign = 64; //  '@'
export const uppercaseA = 65; //  'A'
export const uppercaseB = 66; //  'B'
export const uppercaseC = 67; //  'C'
export const uppercaseD = 68; //  'D'
export const uppercaseE = 69; //  'E'
export const uppercaseF = 70; //  'F'
export const uppercaseG = 71; //  'G'
export const uppercaseH = 72; //  'H'
export const uppercaseI = 73; //  'I'
export const uppercaseJ = 74; //  'J'
export const uppercaseK = 75; //  'K'
export const uppercaseL = 76; //  'L'
export const uppercaseM = 77; //  'M'
export const uppercaseN = 78; //  'N'
export const uppercaseO = 79; //  'O'
export const uppercaseP = 80; //  'P'
export const uppercaseQ = 81; //  'Q'
export const uppercaseR = 82; //  'R'
export const uppercaseS = 83; //  'S'
export const uppercaseT = 84; //  'T'
export const uppercaseU = 85; //  'U'
export const uppercaseV = 86; //  'V'
export const uppercaseW = 87; //  'W'
export const uppercaseX = 88; //  'X'
export const uppercaseY = 89; //  'Y'
export const uppercaseZ = 90; //  'Z'
export const leftSquareBracket = 91; //  '['
export const backslash = 92; //  '\    '
export const rightSquareBracket = 93; //  ']'
export const caret = 94; //  '^'
export const underscore = 95; //  '_'
export const graveAccent = 96; //  '`'
export const lowercaseA = 97; //  'a'
export const lowercaseB = 98; //  'b'
export const lowercaseC = 99; //  'c'
export const lowercaseD = 100; //  'd'
export const lowercaseE = 101; //  'e'
export const lowercaseF = 102; //  'f'
export const lowercaseG = 103; //  'g'
export const lowercaseH = 104; //  'h'
export const lowercaseI = 105; //  'i'
export const lowercaseJ = 106; //  'j'
export const lowercaseK = 107; //  'k'
export const lowercaseL = 108; //  'l'
export const lowercaseM = 109; //  'm'
export const lowercaseN = 110; //  'n'
export const lowercaseO = 111; //  'o'
export const lowercaseP = 112; //  'p'
export const lowercaseQ = 113; //  'q'
export const lowercaseR = 114; //  'r'
export const lowercaseS = 115; //  's'
export const lowercaseT = 116; //  't'
export const lowercaseU = 117; //  'u'
export const lowercaseV = 118; //  'v'
export const lowercaseW = 119; //  'w'
export const lowercaseX = 120; //  'x'
export const lowercaseY = 121; //  'y'
export const lowercaseZ = 122; //  'z'
export const leftCurlyBrace = 123; //  '{'
export const verticalBar = 124; //  '|'
export const rightCurlyBrace = 125; //  '}'
export const tilde = 126; //  '~'
export const nonBreakingSpace = 160;
export const oghamSpaceMark = 5760; // ' '
export const lineSeparator = 8232;
export const paragraphSeparator = 8233;
```
We can add functions like`isWhiteSpace` to it:

```javascript
export function isWhiteSpace(code) {
  return code === lineFeed || space || carriageReturn || tab;
}
``` 
or for digits define a function `isDigit`:

```javascript
export function isDigit(code) {
  return code >= digit0 && code <= digit9;
}
```
But I think this is out of the scope of this PR. I'll apply this in another PR so we can discuss more about it in details there. For now I think what you suggested covers this PR.

**Pull request merged**